tls_dialer.go

```diff
@@ -0,0 +1,60 @@
+package zk
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"net"
+	"os"
+	"time"
+)
+
+func CreateTLSConfig(rootCAFile, certFile, keyFile string) (*tls.Config, error) {
```

Comment:
This should not assume files, and have `io.Reader` as input for all three inputs. This allows a better library experience having tests and callers pass in content as they see fit. 

also the tls package allows this to work then with https://pkg.go.dev/crypto/tls#X509KeyPair